scene/gui/control.cpp

```diff
@@ -1791,19 +1791,28 @@ Control::FocusMode Control::get_focus_mode() const {
 	return data.focus_mode;
 }
 
+void Control::set_focus_layer(int layer_id) {
+	data.focus_layer = layer_id;
```

Comment:
When you change layer of a focused control, it may cause multiple controls on the same layer to be focused. You need to add a check here.